libraries/AP_Filesystem/AP_Filesystem_ROMFS.h

```diff
@@ -17,18 +17,16 @@
 
 #include "AP_Filesystem_backend.h"
 
-#if HAVE_FILESYSTEM_SUPPORT
-
 class AP_Filesystem_ROMFS : public AP_Filesystem_Backend
 {
 public:
     // functions that closely match the equivalent posix calls
     int open(const char *fname, int flags) override;
     int close(int fd) override;
-    ssize_t read(int fd, void *buf, size_t count) override;
-    ssize_t write(int fd, const void *buf, size_t count) override;
+    int32_t read(int fd, void *buf, uint32_t count) override;
+    int32_t write(int fd, const void *buf, uint32_t count) override;
     int fsync(int fd) override;
-    off_t lseek(int fd, off_t offset, int whence) override;
+    int32_t lseek(int fd, int32_t offset, int whence) override;
```

Comment:
Why change all of these?

Reply:
because off_t not available on systems that don't have FATFS, which meant we couldn't build ROMFS, param etc on those boards
also, we normally use stdint types in ardupilot, these were the anomoly